moduleGenerator.py

```diff
@@ -1,36 +1,51 @@
-spline_context.insert_function(eval_spline_2d_scalar, ['double','double','double[:]','int','double[:]','int','double[:,:]','int','int'])
-spline_context.insert_function(eval_spline_1d_scalar, ['double','double[:]','int','double[:]','int'])
-spline_context.insert_function(eval_spline_1d_vector, ['double','double[:]','int','double[:]','double[:]','int'])
-
-init_context = ContextPyccel(name='initialiser_funcs',context_folder='pygyro.initialisation',output_folder='pygyro.advection')
-init_context.insert_function(fEq, ['double','double','double','double','double','double','double','double','double'])
-
-aas_func = epyccel(pygyro.advection.accelerated_advection_steps, context=[spline_context,init_context])
+from pygyro.splines.spline_eval_funcs import find_span, basis_funs, basis_funs_1st_der
+
+spline_context = ContextPyccel(
+    name='context_1', context_folder='pygyro.splines', output_folder='pygyro.splines')
+spline_context.insert_function(find_span, ['double[:]', 'int', 'double'])
+spline_context.insert_function(
+    basis_funs, ['double[:]', 'int', 'double', 'int', 'double[:]'])
+spline_context.insert_function(
+    basis_funs_1st_der, ['double[:]', 'int', 'double', 'int', 'double[:]'])
+
+spline_eval_funcs = epyccel(
+    pygyro.splines.spline_eval_funcs, context=spline_context)
+
+
+init_context = ContextPyccel(
+    name='initialiser_funcs', context_folder='pygyro.initialisation', output_folder='pygyro.initialisation')
+init_context.insert_function(
+    f_eq, ['double', 'double', 'double', 'double', 'double', 'double', 'double', 'double', 'double'])
+init_context.insert_function(perturbation, [
+                             'double', 'double', 'double', 'int', 'int', 'double', 'double', 'double'])
+
+initialiser_func = epyccel(
+    pygyro.initialisation.initialiser_func, context=init_context)
+
+
+spline_context = ContextPyccel(
+    name='spline_eval_funcs', context_folder='pygyro.splines', output_folder='pygyro.advection')
+spline_context.insert_function(eval_spline_2d_cross, [
+                               'double[:]', 'double[:]', 'double[:]', 'int', 'double[:]', 'int', 'double[:,:]', 'double[:,:]', 'int', 'int'])
+spline_context.insert_function(eval_spline_2d_scalar, [
+                               'double', 'double', 'double[:]', 'int', 'double[:]', 'int', 'double[:,:]', 'int', 'int'])
+spline_context.insert_function(eval_spline_1d_scalar, [
+                               'double', 'double[:]', 'int', 'double[:]', 'int'])
+spline_context.insert_function(eval_spline_1d_vector, [
+                               'double', 'double[:]', 'int', 'double[:]', 'double[:]', 'int'])
+
+init_context = ContextPyccel(
+    name='initialiser_funcs', context_folder='pygyro.initialisation', output_folder='pygyro.advection')
+init_context.insert_function(
+    f_eq, ['double', 'double', 'double', 'double', 'double', 'double', 'double', 'double', 'double'])
+
+aas_func = epyccel(pygyro.advection.accelerated_advection_steps,
```

Comment:
This file can be deleted. It was needed for pyccel in early 2018 but was already obsolete by the summer

Reply:
ok, will do